hal/src/sercom/v2/uart/mod.rs

```diff
@@ -0,0 +1,1005 @@
+//! Use the SERCOM peripheral for UART communications
```

Comment:
By the way, I plan to propose that we move away from using `mod.rs`. I have a few reasons:

1. When you have multiple `mod.rs` files open in the same editor, it becomes difficult to know which file you are looking at. Taken to it's extreme, this approach would name every single module `mod.rs`, which doesn't really make sense.
2. If you have a preexisting module without any sub-modules and you add a new sub-module, the `mod.rs` approach requires that you rename the module and move it to a sub-directory. The other approach does not.
3. With the new approach, the directory hierarchy exactly matches the module hierarchy. Sub-modules of a given module are always located one directory level below the module.

The only downside is that it requires a slight change in perspective. Whereas the `uart/` directory *used* to contain the **contents** of the module, it now only contains the **sub-modules** of the module. But once you get used to it, I find that approach much more intuitive.

I discussed this issue with @vccggorski and @vcchtjader, and they agreed with it at the time. I haven't discussed it with anyone else though. I should probably make an issue.

The HAL mixes both approaches right now, so it's not a priority to change it. But I thought I should bring it up, since you took the time to separate out your module into neat little pieces. What do you think?

Reply:
I don't really have preference either way. At first I thought it was weird that module files were called `mod.rs`. I got used to it, so I don't really give it much thought anymore. Though I can definitely see the argument for avoiding `mod.rs`. I'll update the PR to reflect that.